packages/client/src/components/UploadForm.tsx

```diff
@@ -189,14 +171,52 @@ export const UploadForm = () => {
 			return;
 		}
 
+		setStatus(RequestStatus.InProgress);
+
+		if (mediaSource === 'url') {
+			const urls = mediaUrlText.split('\n').filter((url) => url !== '');
+			if (urls.length === 0) {
+				return;
+			}
+			const urlsWithStatus = Object.fromEntries(
+				urls.map((url) => [
+					url,
+					checkUrlValid(url) ? RequestStatus.InProgress : RequestStatus.Invalid,
```

Comment:
Agreed this would be better done prior to submission. This could be done with this nice validation stuff in flowbite https://flowbite-react.com/docs/components/forms#form-validation 

The reason I haven't done it is because at the moment I'm just using a single text area for all the urls, so it would be tricky to indicate which one is invalid via the form element. I think a better solution would be to have one text field per url, and a button to add an extra field, but it's a fair bit of extra work so was going to launch with this as is